### ISSUE TITLE
Update existing Ginger and Gingerbase objstores with 'Version' field

### ISSUE THREAD

**Issue by danielhb** [kimchi-project/kimchi#764]
After the patch set "Upgrade ObjectStore data to Kimchi-2.0" went upstream, existing Ginger objstore files fails to load with the following error:

```
[23/Nov/2015:10:45:18] ENGINE Started monitor thread 'Autoreloader'.
[23/Nov/2015:10:45:19] ENGINE Serving on http://127.0.0.1:37848
[23/Nov/2015:10:45:19] ENGINE Bus STARTED
Traceback (most recent call last):
  File "/home/danielhb/kimchi/wok/kimchi/src/wok/asynctask.py", line 65, in _save_helper
    session.store('task', self.id, obj)
  File "/home/danielhb/kimchi/wok/kimchi/src/wok/objectstore.py", line 92, in store
    (ident, obj_type, jsonstr, version))
OperationalError: table objects has no column named version

WOKASYNC0002E: WOKASYNC0002E 
```

The changes made in Kimchi object store must be done in Ginger and Gingerbase object stores as well.

**Comment by pvital** [comment]
@danielhb: please, tell me how exactly you are testing wok+ginger to get this error, because I could not see this issue when testing with upstream code (kimchi: f28a30e and ginger: 648366e).

**Comment by danielhb** [comment]
I've noticed this error when trying to fix the task API for Ginger (https://github.com/kimchi-project/ginger/issues/37). There is also a patch set from @pkulkark  that presented the same error because she tried to use the Task api with the objstore.

To make it easier for you to debug it, I've created a simple .py file to cause the error:

```
from wok import config
from wok.objectstore import ObjectStore
from wok.utils import add_task

def a_simple_task():
    print 'a simple task'

objstore_loc = config.get_object_store() + '_ginger'
objstore = ObjectStore(objstore_loc)

task_params = {}
taskid = add_task(u'/simple/task/url', a_simple_task, objstore, task_params)

print 'taskid = ', taskid
```

Save it as 'objstore_bug.py' in ginger project root. Then run it with

``` sudo PYTHONPATH=../../../ python objstore_bug.py ```

This is the full output I get in my machine:

```
[danielhb@arthas ginger]$ sudo PYTHONPATH=../../../ python objstore_bug.py 
Traceback (most recent call last):
  File "/home/danielhb/kimchi/wok/kimchi/src/wok/asynctask.py", line 65, in _save_helper
    session.store('task', self.id, obj)
  File "/home/danielhb/kimchi/wok/kimchi/src/wok/objectstore.py", line 92, in store
    (ident, obj_type, jsonstr, version))
OperationalError: table objects has no column named version

WOKASYNC0002E: Unable to start task due error: table objects has no column named version
Traceback (most recent call last):
  File "objstore_bug.py", line 12, in <module>
    taskid = add_task(u'/simple/task/url', a_simple_task, objstore, task_params)
  File "/home/danielhb/kimchi/wok/kimchi/src/wok/utils.py", line 53, in add_task
    AsyncTask(id, target_uri, fn, objstore, opaque)
  File "/home/danielhb/kimchi/wok/kimchi/src/wok/asynctask.py", line 41, in __init__
    self._save_helper()
  File "/home/danielhb/kimchi/wok/kimchi/src/wok/asynctask.py", line 67, in _save_helper
    raise OperationFailed('WOKASYNC0002E', {'err': e.message})
wok.exception.OperationFailed: WOKASYNC0002E: Unable to start task due error: table objects has no column named version
[danielhb@arthas ginger]$ 
```

**Comment by pvital** [comment]
@danielhb please, paste here the schema of this Ginger objectstore file. Do the follow:

```
$ sqlite3 <objectstore file>
sqlite> .schema objects
```

**Comment by danielhb** [comment]
```
[danielhb@arthas kimchi]$ cd data/
[danielhb@arthas data]$ ls
objectstore  objectstore_ginger  objectstore_gingerbase
[danielhb@arthas data]$ sqlite3 objectstore_ginger
SQLite version 3.8.10.2 2015-05-20 18:17:19
Enter ".help" for usage hints.
sqlite> .schema objects
CREATE TABLE objects
                (id TEXT, type TEXT, json TEXT, PRIMARY KEY (id, type));
sqlite> 
```

**Comment by pvital** [comment]
Ok, the problem is: the schema used to create the objectstore_ginger file is the old schema. 
It's necessary do the same as I did to Kimchi objectstore file: check the schema and update it if necessary. Will work on the solution and send a patch!

@danielhb could you send me in private this file you have? Thanks!

**Comment by danielhb** [comment]
@pvital I've sent both ginger and gingerbase objectstore files I have here

Let me know if you need any assistance in solving this.

**Comment by danielhb** [comment]
Update: after @pkulkark saying that she couldn't reproduce the bug, I've tried to remove the existing objstore files and recreate them (make clean; ./autogen.sh --system; make). The issue seems solved.

This means that the issue apparently does not happen with new objestore files. Just with existing ones.

**Comment by pvital** [comment]
@danielhb yeah! This happens because the schema of the objectstore files was changed from kimchi 1.5.1 to wok/kimchi 2.0 and when you don't have the objectstore file, a new one will be created with the new schema.

It's necessary develop a script to update the Ginger and GingerBase objectstore files if necessary, as same as I did for Kimchi file. When the plugin is loaded the update script is executed if necessary.

I'm going to see this and send a solution ASAP.
